core/src/main/java/edu/wpi/grip/core/sources/CameraSource.java

```diff
@@ -154,8 +154,16 @@ public void start() throws IOException, IllegalStateException {
 
                     frameMat.copyTo(frameOutputSocket.getValue().get());
                     frameOutputSocket.setValue(frameOutputSocket.getValue().get());
-                    long thisMoment = System.currentTimeMillis();
-                    frameRateOutputSocket.setValue(1000 / (thisMoment - lastFrame));
+
+                    final long thisMoment = System.currentTimeMillis();
+                    final long elapsedTime = thisMoment - lastFrame;
+                    if (elapsedTime == 0) {
+                        // This can actually happen: https://github.com/WPIRoboticsProjects/GRIP/issues/199
+                        frameRateOutputSocket.setValue(Double.NaN);
```

Comment:
I think we should try not to have NaNs in the pipeline to avoid weird unexpected behavior.  A decent way to handle this would probably be to not change `frameRateOutputSocket` if essentially no time has elapsed, or to set it to 0.

These other options aren't technically mathematically correct, but I think it would be less suprising and less likely to break things.